src/python/pants/backend/python/rules/pytest_runner.py

```diff
@@ -234,15 +234,15 @@ async def run_python_test(
 
     use_coverage = test_options.values.use_coverage
     output_dirs = [".coverage"] if use_coverage else []
-    if test_setup.xml_dir:
-        output_dirs.append(test_results_file)
+    output_files = [test_results_file] if test_setup.xml_dir else []
```

Comment:
We might ad well do the same for `. coverage` while we make this change since it is seems to be busted for RE too, see https://github.com/pantsbuild/pants/pull/9919

Reply:
`.coverage` is a file?

Reply:
yes.